templates/kong.yml.erb

```diff
@@ -1,394 +1,102 @@
-  # encrypt: "foo"
+<% @dns_resolvers_available.each do |resolver, values| -%>
+  <%= resolver -%>:
+<% values.each do |k,v| -%>
+    <%= k %>: "<%= v %>"
+<% end -%>
+<% end -%>
 
+<% if @cluster_advertise || @cluster_encrypt || @cluster_ttl_on_failure -%>
 cluster:
-<% if not @cluster_advertise.nil? and @cluster_advertise -%>
-  '<%= @cluster_advertise %>'
+<% if @cluster_advertise -%>
+  advertise: "<%= @cluster_advertise %>"
+<% end -%>
+<% if @cluster_encrypt -%>
+  encrypt: "<%= @cluster_encrypt %>"
+<% end -%>
+<% if @cluster_ttl_on_failure -%>
+  ttl_on_failure: "<%= @cluster_ttl_on_failure %>"
 <% end -%>
-<% if not @cluster_encrypt.nil? and @cluster_encrypt -%>
-  "<%= @cluster_encrypt %>"
 <% end -%>
 
-######
-## Specify which database to use. Only "cassandra" is currently available.
-# database: cassandra
-
-database: '<%= @database -%>'
-
-######
-## Cassandra configuration (keyspace, authentication, client-to-node encryption)
-# cassandra:
-  ######
-  ## Contact points to your Cassandra cluster.
-  # contact_points:
-  #   - "127.0.0.1:9042"
-
-  ######
-  ## Name of the keyspace used by Kong. Will be created if it does not exist.
-  # keyspace: kong
-
-  #####
-  ## Connection and reading timeout (in ms).
-  # timeout: 5000
-
-  ######
-  ## Keyspace options. Set those before running Kong or any migration.
-  ## Those settings will be used to create a keyspace with the desired options
-  ## when first running the migrations.
-  ## See http://docs.datastax.com/en/cql/3.1/cql/cql_reference/create_keyspace_r.html
-  ######
-  ## The name of the replica placement strategy class for the keyspace.
-  ## Can be "SimpleStrategy" or "NetworkTopologyStrategy".
-  # replication_strategy: SimpleStrategy
-  ######
-  ## For SimpleStrategy only.
-  ## The number of replicas of data on multiple nodes.
-  # replication_factor: 1
-  ######
-  ## For NetworkTopologyStrategy only.
-  ## The number of replicas of data on multiple nodes in each data center.
-  # data_centers:
-  #   dc1: 2
-  #   dc2: 3
-
-  #####
-  ## Client-to-node TLS options.
-  ## `enabled`: if true, will connect to your Cassandra instance using TLS.
-  ## `verify`: if true, will verify the server certificate using the given CA file.
-  ## `certificate_authority`: an absolute path to the trusted CA certificate in PEM format used to verify the server certificate.
-  ## For additional SSL settings, see the ngx_lua `lua_ssl_*` directives.
-  # ssl:
-  #   enabled: false
-  #   verify: false
-  #   certificate_authority: "/path/to/cluster-ca-certificate.pem"
-
-  ######
-  ## Cluster authentication options. Provide a user and a password here if your cluster uses the
-  ## PasswordAuthenticator scheme.
-  # username: cassandra
-  # password: cassandra
+database: "<%= @database -%>"
 
+<% if @database == 'postgres' -%>
+postgres:
+  host: "<%= @postgres_host -%>"
+  port: "<%= @postgres_port -%>"
```

Comment:
Get rid of the quotes here. This needs to be a number, not a string, or Kong does not start up.